src/main/java/org/commcare/formplayer/application/MenuController.java

```diff
@@ -140,6 +150,9 @@ public EntityDetailListResponse getDetails(@RequestBody SessionNavigationBean se
             throw new RuntimeException("Could not find case with ID " + detailSelection);
         }
 
+        String cacheKey = UserUtils.getFullUserDetail(sessionNavigationBean.getUsername(), sessionNavigationBean.getRestoreAs(), sessionNavigationBean.getDomain());
+        String cacheValue = String.join("|", selections);
```

Comment:
These `String.join` calls should be abstracted into a call somewhere that has clearer semantic meaning (IE: getSessionCacheKey(selections)

I wonder if it would be worth giving the "selections" array here its own business logic object (and potentially the other items above, IE: 'commitSelections' 'detailSeleciton' etc) given how many places we seem to be making assumptions about multiple arrays of strings and what their semantic meaning is. 

Depending on where that object comes from in the lifecycle, you might be able to build more robust cache keys around it in the future.